### PR TITLE
AArch64: Add space for outgoing JNI argument to J9CInterpreterStackFrame

### PR DESCRIPTION
**Pull request by Akira1Saitoh**
This commit adds outgoingArguments array to J9CInterpreterStackFrame for AArch64 VM to avoid overwriting preservedGPRs and preservedFPRs when calling out to jitReleaseVMAccess helper from JNI invocation sequence.

This commit also updates ARM64JNILinkage to store JNI method arguments that are passed by stack to the area where the outgoingArguments array resides.

Adding outgoingArguments makes the size of J9CInterpreterStackFrame large and some slots are not accessible with ldp/stp instructions using sp as a base register. Thus, this commit also updates m4 macros in arm64helpers.m4. They use the start of slots for saving JIT GPRs as the base address and x16 as the base register.

Issue: https://github.com/eclipse-openj9/openj9/issues/18149

### CODE REVIEW

File: runtime/compiler/aarch64/codegen/ARM64JNILinkage.cpp [hunk 1]
```diff
@@ -503,12 +502,11 @@ size_t J9::ARM64::JNILinkage::buildJNIArgs(TR::Node *callNode, TR::RegisterDepen
    if (numMemArgs > 0)
       {
       pushToMemory = new (trStackMemory()) TR::ARM64MemoryArgument[numMemArgs];
-
-      argMemReg = cg()->allocateRegister();
       }
 
    // align to 16-byte boundary
    totalSize = (totalSize + 15) & (~15);
+   TR_ASSERT_FATAL(totalSize <= 256, "totalSize must not be larger than 256");
```
Comment (knn-k) [inline runtime/compiler/aarch64/codegen/ARM64JNILinkage.cpp:509]: Please add a comment for the reason of this size limit.

File: runtime/compiler/aarch64/codegen/ARM64JNILinkage.cpp [hunk 2]
```diff
@@ -1059,19 +1053,6 @@ TR::Register *J9::ARM64::JNILinkage::buildDirectDispatch(TR::Node *callNode)
    TR::RegisterDependencyConditions *deps = new (trHeapMemory()) TR::RegisterDependencyConditions(maxRegisters, maxPostRegisters, trMemory());
 
    size_t spSize = buildJNIArgs(callNode, deps, passThread, passReceiver, killNonVolatileGPRs);
```
Comment (knn-k) [inline runtime/compiler/aarch64/codegen/ARM64JNILinkage.cpp:1055]: `spSize` will be unused after the change.

File: runtime/compiler/aarch64/codegen/ARM64JNILinkage.cpp (hunk 1 above)
Comment (Akira1Saitoh) [reply runtime/compiler/aarch64/codegen/ARM64JNILinkage.cpp:516]: Added a comment.

File: runtime/compiler/aarch64/codegen/ARM64JNILinkage.cpp (hunk 2 above)
Comment (Akira1Saitoh) [reply runtime/compiler/aarch64/codegen/ARM64JNILinkage.cpp]: Removed `spSize` variable.

**Comment by Akira1Saitoh** [comment]
@gacholio May I ask you to review the changes to the helper in this PR?

**Comment by knn-k** [comment]
The build failure with aarch64_mac will be fixed by PR #18233.

**Comment by knn-k** [comment]
Jenkins test sanity.functional,sanity.system amac jdk17

**Review by gacholio — APPROVED** [review]

**Pull request merged**